components/kyma-environment-broker/internal/process/provisioning/event_hub_step.go

```diff
@@ -96,29 +98,34 @@ func (p *ProvisionAzureEventHubStep) Run(operation internal.ProvisioningOperatio
 		azure.TagOperationID:  &operation.ID,
 	}
 
+	// Generating a unique name for resource group and namespace with operation ID as suffix if available
+	uniqueName, err := p.getUniqueNSName(operation.ID, azureClient, log)
```

Comment:
> mhh ... isn't this a typical [TOCTTOU](https://de.wikipedia.org/wiki/Time-of-Check-to-Time-of-Use-Problem) problem? Instead of checking, we could just generate the first name, try it, check for a conflict on azure side (some 4xx) and repeat with other names if required.

You are correct that indeed a race condition can happen between time of check and time of creation, however the design of the Azure SDK does not return a clear status code/error for this condition (unless you wanna parse an error string and our client breaks in case Azure changes the error message, which I'm really not in favour of).

 Generally, TOCTTOC problems are critical on the service side because say if it was vulnerable it can cause two namespaces to be created with the same name.

However, in our case, there is no harm if a race occurred between TOC and TOC 😄 . All the logic of checking NS availability is just to handle the very corner case of the operation ID which is a (UUID) collision, something that is supposedly a very low probability but safer to handle.

Reply:
> If `getUniqueNSName` fails because of network problems, shouldn't we retry the operation then? There might be a chance that the network problem goes away after a few minutes.
> 
> This is at least how an error during `CreateNamespace` is treated atm.

You are right. Done.